src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql

```diff
@@ -0,0 +1,301 @@
+// valueset "Dementia NOS": 'F03.9'
+// valueset "Vascular Dementia, unspecified severity": 'F01.5'
+// valueset "Frontotemporal dementia": 'G31.0'
+// valueset "Parkinson's disease": 'G20'
+// valueset "Unspecified dementia, with behavioral disturbance": 'F03.91'
+
+// define "Conditions Indicating Dementia":
+//     ["Condition": "Dementia NOS"]
+//     union ["Condition": "Vascular Dementia, unspecified severity"]
+//     union ["Condition": "Frontotemporal dementia"]
+//     union ["Condition": "Parkinson's disease"]
+//     union ["Condition": "Unspecified dementia, with behavioral disturbance"]
+
+define DementiaConditions:
+  (["Condition"] C where (
+    C.code.coding[0].code.value = 'F03.9' or
+    C.code.coding[0].code.value = 'F01.5' or
+    C.code.coding[0].code.value = 'G31.0' or
+    C.code.coding[0].code.value = 'G20' or
+    C.code.coding[0].code.value = 'F03.91'
+  ))
+
+define HasDementia:
+  Length(DementiaConditions) > 0
+
+
+//   depression condition ("F32.A - Depression, unspecified" and "F33.1 - Major Depressive Disorder, Recurrent, Moderate")
+//   and/or 
+//   anxiety condition ("F41.1 - Generalized Anxiety Disorder" - not present in our 3 test patients...)
+
+define MoodConditions: 
+  ["Condition"] C where (
+    C.code.coding[0].code.value = 'F32.A' or
+    C.code.coding[0].code.value = 'F33.1' or
+    C.code.coding[0].code.value = 'F41.1'
+  )
+
+define HasMoodConditions:
+  Length(MoodConditions) > 0
+
+
+// Naming pattern: "CIRG_MINICOG" is Questionnaire.id
+define PresentQnr_CIRG_MINICOG:
+  if dcw_cog_level_1 is true then {
+    id: 'CIRG-MINICOG',
+    schedule: Scheduled_Timing
+  } else null
+
+
+define PresentQnr_CIRG_ECOG12:
+  if dcw_cog_level_1 is true then {
+    id: 'CIRG-ECOG12',
+    schedule: Scheduled_Timing
+  }
+  else null
+
+define PresentQnr_CIRG_CP_ECOG:
+  if dcw_cog_level_1 is true and dcw_care_partner_present is true then {
+    id: 'CIRG-CP-ECOG',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_SLUMS:
+  if dcw_cog_level_2 is true then {
+    id: 'CIRG-SLUMS',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_ADL_IADL:
+  if dcw_cog_level_2 is true and dcw_care_partner_present is true then {
+    id: 'CIRG-ADL-IADL',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_BEHAV5:
+  if dcw_mood_level_1 is true then {
+    id: 'CIRG-BEHAV5',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_GAD7:
+ // TODO figure out what score this is supposed to be
+ // CIRG-SLUMS score is Normal or Mild Neurocognitive Disorder
+ // dependent on education level? using High School Education criteria here 
+  if dcw_mood_level_2 is true and First(SLUMS_Score) >= 21 then {
+    id: 'CIRG-GAD7',
+    schedule: Scheduled_Timing
+  } else null
+
+define PresentQnr_CIRG_GDS:
+  // TODO figure out what score this is supposed to be
+  // CIRG-SLUMS score is Normal or Mild Neurocognitive Disorder
```

Comment:
@sierramatice I am not certain what SLUMS score is normal or indicative of mild neurocognitive Disorder.  I _think_ it may be dependent on education level if I remember correctly.  But let me know?

Reply:
The SLUMS scoring is based on whether the patient has a high school diploma or not. 

The first place I looked for education information is US Core (data elements required to be captured in the EHR). That includes social determinants of health (https://www.hl7.org/fhir/us/core/sdoh.html ).
From that link, "SDOH Problems/Health Concerns: Identifying a Social Determinants of Health-related condition (for example, homelessness) is represented by the [US Core Condition Problems and Health Concerns Profile](https://www.hl7.org/fhir/us/core/StructureDefinition-us-core-condition-problems-health-concerns.html)".

The only education-related element there is "Less than a high school diploma", which is what we need. This is considered a condition, and it's coding is:
https://cts.nlm.nih.gov/fhir/CodeSystem/$lookup?system=http://hl7.org/fhir/sid/icd-10-cm&version=2023&code=Z55.5 

So @achen2401  I think we could represent this as a Condition with that coding, and that it's absence would mean they have a higher education level than that.

I expect there's quite a few ways people go about modelling this - the above approach isn't very flexible, but it does fit our need here.

@ivan-c  fyi.

Reply:
@achen2401 it looks like you didn't implement "Less than a high school diploma" as a FHIR Condition, can you please clarify? @sierramatice  noted that Ross Smith says the cutoff score for SLUMS seems incorrect perhaps due to that.

Reply:
@mcjustin  Hmm, I did implement that.  Please see [here](https://github.com/uwcirg/asbi-screening-app/blob/a6d389c78e7696d037cdf31cc809f4db8cf19f2e/src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql#L380).   Not the SLUMS scores are evaluated based on education level See [here](https://github.com/uwcirg/asbi-screening-app/blob/a6d389c78e7696d037cdf31cc809f4db8cf19f2e/src/cql/upstream-external/CirgLibraryQuestionnaireLogic_DCW.cql#L270).    Unless the logic and/or score cutoff is incorrect somewhere.  Let me know?

Reply:
Great, then it's just that we're missing Ross Smith's condition for this - I'll add it (I have a set of new patients to add as well).

Reply:
Done w/ https://github.com/uwcirg/asbi-screening-app/pull/79